packages/swi_prolog.rb

```diff
@@ -0,0 +1,58 @@
+require 'package'
+require 'fileutils'
+
+class Swi_prolog < Package
+  description 'SWI-Prolog offers a comprehensive free Prolog environment. Since its start in 1987, SWI-Prolog development has been driven by the needs of real world applications. SWI-Prolog is widely used in research and education as well as commercial applications. Join over a million users who have downloaded SWI-Prolog.'
+  homepage 'https://www.swi-prolog.org/'
+  version '8.2.4'
+  compatibility 'all'
+  source_url 'https://www.swi-prolog.org/download/stable/src/swipl-8.2.4.tar.gz'
+  source_sha256 'f4bcc78437f9080ab089762e9e6afa7071df7f584c14999b92b9a90a4efbd7d8'
+
+  binary_url ({
+  })
+  binary_sha256 ({
+  })
+
+  depends_on 'zlibpkg'
+  depends_on 'openssl'
+  depends_on 'gmp'
+  depends_on 'readline'
+  depends_on 'libarchive'
+  depends_on 'ncurses'
+  depends_on 'libdb'
+  depends_on 'acl'
+  depends_on 'ninja' => :build
+  depends_on 'cmake' => :build
+  depends_on 'llvm' => :build
+
+  def self.build
+    FileUtils.remove_dir 'build', true
+    Dir.mkdir 'build'
+    Dir.chdir 'build' do
+        system "cmake -G Ninja \
+        #{CREW_CMAKE_OPTIONS} \
+        -DCURSES_CURSES_LIBRARY=/usr/local/lib64/libncurses.so.6 \
+        -DCURSES_INCLUDE_PATH=/usr/local/include/ncurses  \
+        -DCMAKE_LIBRARY_PATH=/usr/local/lib64 \
```

Comment:
Please use crew constants to abstract instead of hardcoded directories.  For example, `/usr/local/lib64` should be `#{CREW_LIB_PREFIX}` instead since that directory only exists on x86_64 architecture.  Actually, you don't need `-DCMAKE_LIBRARY_PATH=/usr/local/lib64` since that is already included in `#{CREW_CMAKE_OPTIONS`.

Reply:
Good catch. Sorry I missed that. I probably copy-pasted from my manual compilation and forgot to fix that up.

Reply:
Updated.